tests/plugins/UM3NetworkPrinting/TestSendMaterialJob.py

```diff
@@ -0,0 +1,314 @@
+    def __init__(self):
+        self.containersMetaData = None
+
+    def findContainers(self, *, ignore_case: bool = False, **kwargs: Any) -> List[ContainerInterface]:
+        raise NotImplementedError()
+
+    def findDefinitionContainers(self, **kwargs: Any) -> List[DefinitionContainerInterface]:
+        raise NotImplementedError()
+
+    @classmethod
+    def getApplication(cls) -> "Application":
+        raise NotImplementedError()
+
+    def getEmptyInstanceContainer(self) -> "InstanceContainer":
+        raise NotImplementedError()
+
+    def isReadOnly(self, container_id: str) -> bool:
+        raise NotImplementedError()
+
+    def setContainersMetadata(self, value):
+        self.containersMetaData = value
+
+    def findContainersMetadata(self, type):
+        return self.containersMetaData
+
+
+class FakeDevice(ClusterUM3OutputDevice):
+    def _createFormPart(self, content_header, data, content_type=None):
+        return "xxx"
+
+
+class TestSendMaterialJob(TestCase):
+    _LOCALMATERIAL_WHITE = {'type': 'material', 'status': 'unknown', 'id': 'generic_pla_white',
+                            'base_file': 'generic_pla_white', 'setting_version': 5, 'name': 'White PLA',
+                            'brand': 'Generic', 'material': 'PLA', 'color_name': 'White',
+                            'GUID': 'badb0ee7-87c8-4f3f-9398-938587b67dce', 'version': '1', 'color_code': '#ffffff',
+                            'description': 'Test PLA White', 'adhesion_info': 'Use glue.', 'approximate_diameter': '3',
+                            'properties': {'density': '1.00', 'diameter': '2.85', 'weight': '750'},
+                            'definition': 'fdmprinter', 'compatible': True}
+    _LOCALMATERIAL_BLACK = {'type': 'material', 'status': 'unknown', 'id': 'generic_pla_black',
+                            'base_file': 'generic_pla_black', 'setting_version': 5, 'name': 'Yellow CPE',
+                            'brand': 'Ultimaker', 'material': 'CPE', 'color_name': 'Black',
+                            'GUID': '5fbb362a-41f9-4818-bb43-15ea6df34aa4', 'version': '1', 'color_code': '#000000',
+                            'description': 'Test PLA Black', 'adhesion_info': 'Use glue.', 'approximate_diameter': '3',
+                            'properties': {'density': '1.01', 'diameter': '2.85', 'weight': '750'},
+                            'definition': 'fdmprinter', 'compatible': True}
+
+    _REMOTEMATERIAL_WHITE = {
+        "guid": "badb0ee7-87c8-4f3f-9398-938587b67dce",
+        "material": "PLA",
+        "brand": "Generic",
+        "version": 1,
+        "color": "White",
+        "density": 1.00
+    }
+    _REMOTEMATERIAL_BLACK = {
+        "guid": "5fbb362a-41f9-4818-bb43-15ea6df34aa4",
+        "material": "PLA",
+        "brand": "Generic",
+        "version": 2,
+        "color": "Black",
+        "density": 1.00
+    }
+
+    def setUp(self):
+        # Make sure the we start with clean (log) slate
+        _logentries.clear()
```

Comment:
new_log and _logentries seem only used in this test class, so it could just be a property on the class instead of a global one.